### PR TITLE
feat: add nocloud platform support

### PR DESCRIPTION
**Pull request by sergelogvinov**
https://cloudinit.readthedocs.io/en/latest/topics/datasources/nocloud.html#datasource-nocloud

This PR helps to use some automatisation for Proxmox and other custom kvm-clouds

* cdrom version
* net version

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4208)

### CODE REVIEW

File: internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/nocloud.go [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+	}
+
+	return metaConfig, networkConfig, machineConfig, nil
+}
+
+//nolint:gocyclo
+func (n *Nocloud) configFromCD() (metaConfig []byte, networkConfig []byte, machineConfig []byte, err error) {
+	var dev *probe.ProbedBlockDevice
+
+	dev, err = probe.GetDevWithFileSystemLabel(strings.ToLower(configISOLable))
```
Comment (smira) [inline internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/nocloud.go:419]: nit:

```suggestion
	dev, err = probe.GetDevWithFileSystemLabel(strings.ToLower(configISOLabel))
```

Comment (sergelogvinov) [reply internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/nocloud.go]: oh... fixed

**Review by smira — APPROVED** [review]

**Review by rsmitty — APPROVED** [review]
@sergelogvinov doesn't have to be in this PR, but it'd be awesome if you could add a bit to the docs about what NoCloud is, when you would use the nocloud image, and maybe how you'd do it w/ proxmox or something.

Thanks!

**Pull request merged**